dhis-2/dhis-support/dhis-support-system/src/main/java/org/hisp/dhis/cache/DefaultCacheProvider.java

```diff
@@ -646,4 +647,15 @@ public <V> Cache<V> createSubExpressionCache()
             .forRegion( Region.subExpressionCache.name() )
             .expireAfterWrite( 5, TimeUnit.MINUTES ) );
     }
+
+    @Override
+    public <V> Cache<V> createQueryShortenerCache()
+    {
+        return registerCache( this.<V> newBuilder()
+            .forRegion( Region.queryShortenerCache.name() )
+            .expireAfterWrite( 1, TimeUnit.HOURS )
+            .withInitialCapacity( (int) getActualSize( SIZE_100 ) )
+            .forceInMemory()
+            .withMaximumSize( orZeroInTestRun( getActualSize( SIZE_1K ) ) ) );
```

Comment:
Would it make sense not to have an expiry time? And I think if the only thing we are storing in the cache is the hash and original URL + params, you can easily bump it from a 1000 capacity to 10000 (not sure how many are needed).

And I guess these are potentially generated at certain points pr. app right? So it doesn't make sense that an app need to re-create the alias if the app hasn't been used for 1 hour?

Reply:
I think that could make sense @stian-sandvold - the client won't know that the alias exists in many cases, but we could maybe cache it across browser refreshes if we really wanted to try to avoid the extra request.

I also think we should set a maximum entry size and enforce it in the controller

Reply:
@stian-sandvold I removed the expiration and bumped capacity to 10000, also introduced a max entry size of 16kb.  This means the theoretical max size is 160MB which is quite large but should never be reached in practice... thoughts?

Reply:
I'm a bit torn about whether or not to include expiry, I think it's probably fine to have entries expire and in practice they will be repeated recreated anyway.  That would also mean if a client fills up a ton of memory by creating aliases for minor permutations of a really long query string that memory will be automatically cleared, since the entries will expire.  If no clients ever make that request again it would be a waste to keep that in memory forever.

So in thinking about this a little more I think we should keep the expiration, adding it back in now.

Reply:
Setting the expiration time to 3 hours as a compromise.